File: jewels-and-stones/jewels-and-stones.cpp
class Solution {
public:
    int numJewelsInStones(string jewels, string stones) {
        ios_base::sync_with_stdio(false); cin.tie(NULL);
        unordered_map<char, int> umap;
        for (char& c: jewels) umap[c]++;
        int count = 0;
        for (char& c: stones) {
            if (umap[c] != 0)
                count++;
        }
        return count;
    }
};

File: jewels-and-stones/jewels-and-stones.java
class Solution {
    public int numJewelsInStones(String jewels, String stones) {
        HashMap<Character, Integer> hMap = new HashMap<>();
        for (int i=0; i<jewels.length(); ++i)
            hMap.put(jewels.charAt(i), hMap.getOrDefault(jewels.charAt(i), 0) + 1);
        int count = 0;
        for (int i=0; i<stones.length(); ++i)
            if (hMap.get(stones.charAt(i)) != null)
                count++;
        return count;
    }
}